pkg/eks/api.go

```diff
@@ -248,7 +248,7 @@ func (c *ClusterProvider) EnsureAMI(version string, ng *api.NodeGroup) error {
 	}
 
 	// Check the AMI is available
-	available, err := ami.IsAvailable(c.Provider.EC2(), ng.AMI)
+	available, devName, devType, err := ami.IsAvailable(c.Provider.EC2(), ng.AMI)
```

Comment:
Since this is the only place `ami.IsAvailable` is called, how about renaming it and moving everything there?
So here we will have `return ami.Use(c.Provider.EC2(), ng)`, and let that function populate all the fields in `ng` or produce whatever errors it may?

Reply:
I have now done this.